domain/secret/state/state.go

```diff
@@ -1132,6 +1132,37 @@ func (st State) GetSecret(ctx context.Context, uri *coresecrets.URI) (*coresecre
 	return secrets[0], nil
 }
 
+// GetLatestRevision returns the latest revision number for the specified secret,
+// returning an error satisfying [secreterrors.SecretNotFound] if the secret does not exist.
+func (st State) GetLatestRevision(ctx context.Context, uri *coresecrets.URI) (int, error) {
+	db, err := st.DB()
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	query := `
+SELECT MAX(sr.revision) AS &secretInfo.latest_revision
+FROM   secret_revision sr
+WHERE  sr.secret_id = $secretInfo.secret_id
+`
+	info := secretInfo{
+		ID: uri.ID,
+	}
+	stmt, err := st.Prepare(query, info)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, info).Get(&info)
+		if info.LatestRevision == 0 {
```

Comment:
Any error will cause `info.LatestRevision == 0`. We should handle `ErrNoRows` instead.

Reply:
NoRows is never returned - there's always a result

Reply:
But I moved the error check up higher